components/butterfly/src/server/mod.rs

```diff
@@ -1168,16 +1169,34 @@ impl fmt::Display for Server {
 }
 
 fn persist_loop(server: &Server) {
-    // TODO: Make this configurable with EnvConfig. That trait needs to move
-    // to common or core first
-    const MIN_LOOP_PERIOD: Duration = Duration::from_secs(30);
+    struct PersistLoopPeriod(Duration);
```

Comment:
We've got so many of these specialized duration types now (both in this PR and elsewhere) that it might be worth creating a macro that can generate all these implementations for us.

It would ensure consistency if nothing else.

(Just a thought, and not necessarily for this PR.)

Reply:
I had the same thought

Reply:
See https://github.com/habitat-sh/habitat/pull/6565/commits/419d60dbe29be4c931108a60fd36f1f51574fd6a